### PR TITLE
Improves Error Deserialization to dynamically handle edge cases that …

### PR DESCRIPTION
**Pull request by jchen293**
# Description

Improves Error Deserialization to dynamically handle edge cases that have a bad format
# Testing

Add unit tests that have bad format

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -2,79 +2,21 @@
  * A class containing constants used throughout the EasyPost Node.js client library.
  */
 export default class Constants {
-  /**
-   * An array of carrier account types that support custom workflows.
-   * @returns {string[]}
-   */
```
Comment (Justintime50) [inline src/constants.js:8]: praise: thanks for getting rid of these redundant docstrings

File: src/errors/error_handler.js [hunk 2]
```diff
@@ -11,8 +11,30 @@ import InternalServerError from './api/internal_server_error';
...
+  /**
+   * Recursively traverses a JSON object or array and extracts error messages
+   * as strings. Adds the extracted messages to the specified messagesList array.
+   *
+   * @param {object|array|string} errorMessage - The JSON object or array to traverse.
+   * @param {array} messagesList - The array to which extracted error messages will be added.
+   */
+  static traverseJsonElement(errorMessage, messagesList) {
+    if (errorMessage instanceof Object) {
+      for (const value of Object.values(errorMessage)) {
```
Comment (Justintime50) [inline src/errors/error_handler.js:27]: suggestion: we use `value` in one block and `element` in the very next. Let's have them both be the same name

**Review by Justintime50 — DISMISSED** [review]

File: src/errors/error_handler.js (hunk 2 above)
Comment (jchen293) [reply src/errors/error_handler.js:27]: 👌

File: src/errors/error_handler.js [hunk 3]
```diff
@@ -28,39 +50,47 @@ export default class ErrorHandler {
...
+      const errorParams = {
+        message: Constants.ERROR_DESERIALIZATION,
+        code: 'ERROR_DESERIALIZATION_ERROR',
+      };
+      throw new EasyPostError(errorParams);
     }
 
     if (statusCode >= 300 && statusCode < 400) {
-      return new RedirectError(errorParams);
+      throw new RedirectError(errorParams);
```
Comment (nwithan8) [inline src/errors/error_handler.js:66]: This function should not be throwing errors, but rather returning new instances (which are thrown by upstream functions that call this function)

**Review by nwithan8 — CHANGES_REQUESTED** [review]

**Review by Justintime50 — DISMISSED** [review]

File: src/errors/error_handler.js (hunk 3 above)
Comment (jchen293) [reply src/errors/error_handler.js]: That's true. I changed this because the `handleApiError` function isn't throwing an error and can't really test it. I rolled this back and add `throw` in the unit tests. All good.

**Review by nwithan8 — APPROVED** [review]

**Review by Justintime50 — APPROVED** [review]
